src/Compilers/CSharp/Test/Symbol/DocumentationComments/DocumentationCommentCompilerTests.cs

```diff
@@ -949,38 +965,288 @@ partial class C
             // Files passed in order.
             var compA = CreateCompilation(new[] { tree1, tree2 }, assemblyName: "Test");
             var actualA = GetDocumentationCommentText(compA);
-            var expectedA = @"
+            var expected = @"
 <?xml version=""1.0""?>
 <doc>
     <assembly>
         <name>Test</name>
     </assembly>
     <members>
+        <member name=""T:C"">
+            <summary>Summary 0</summary>
+        </member>
         <member name=""M:C.M"">
             <summary>Summary 1</summary>
         </member>
     </members>
 </doc>
 ".Trim();
-            Assert.Equal(expectedA, actualA);
+            AssertEx.Equal(expected, actualA);
 
             // Files passed in reverse order.
             var compB = CreateCompilation(new[] { tree2, tree1 }, assemblyName: "Test");
             var actualB = GetDocumentationCommentText(compB);
-            var expectedB = @"
+            Assert.Equal(expected, actualB);
+        }
+
+        [Fact]
+        public void ExtendedPartialMethods_MultipleFiles_DefinitionComment()
+        {
+            var source1 = @"
+/// <summary>Summary 0</summary>
+public partial class C
+{
+    public partial int M() => 42;
+}
+";
+
+            var source2 = @"
+public partial class C
+{
+    /** <summary>Summary 2</summary>*/
+    public partial int M();
+}
+";
+
+            var tree1 = SyntaxFactory.ParseSyntaxTree(source1, options: TestOptions.RegularWithDocumentationComments);
+            var tree2 = SyntaxFactory.ParseSyntaxTree(source2, options: TestOptions.RegularWithDocumentationComments);
+
+            // Files passed in order.
+            var compA = CreateCompilation(new[] { tree1, tree2 }, assemblyName: "Test");
+            compA.VerifyDiagnostics();
+            var actualA = GetDocumentationCommentText(compA);
+            var expected = @"
+<?xml version=""1.0""?>
+<doc>
+    <assembly>
+        <name>Test</name>
+    </assembly>
+    <members>
+        <member name=""T:C"">
+            <summary>Summary 0</summary>
+        </member>
+        <member name=""M:C.M"">
+            <summary>Summary 2</summary>
+        </member>
+    </members>
+</doc>
+".Trim();
+            AssertEx.Equal(expected, actualA);
+
+            // Files passed in reverse order.
+            var compB = CreateCompilation(new[] { tree2, tree1 }, assemblyName: "Test");
+            compB.VerifyDiagnostics();
+            var actualB = GetDocumentationCommentText(compB);
+            Assert.Equal(expected, actualB);
+        }
+
+        [Fact]
+        public void ExtendedPartialMethods_MultipleFiles_ImplementationComment()
+        {
+            var source1 = @"
+/// <summary>Summary 0</summary>
+public partial class C
+{
+    /** <summary>Summary 1</summary>*/
+    public partial int M() => 42;
```

Comment:
>public partial int M() => 42; [](http://example.com/codeflow?start=4&length=29)

What is the behavior when there are paramref tags in the documentation and parameter names do not match between declaration and implementation? Will emitted method signature match the produced documentation?

Reply:
I'll add some tests for this when I get the chance. If the existing behavior that we observe once tests are in place seems particularly bad, we can consider adjusting it.

Reply:
The behavior I observed is that warnings are given based off the particular declaration whose doc comments we are processing. However, the partial declaration parameter names always "win" in terms of metadata. [SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBBmABABwgCcAXKCAG12VwGFcBvTXF6vQ08qhAFlwFkUAClglcADxQBKANzNWOAsTKVqfQSJhjxSKY1wBfTEYxA==). Potentially, we should consider making it so when we check the validity of 'paramrefs' of a partial implementation, we look to the partial declaration instead for the parameter names.

Partial method parameter name differences are currently a warning, FWIW.

Reply:
> Partial method parameter name differences are currently a warning, FWIW.

I do not see any warnings for the following code:
```
partial class Test
{
    partial void M(int x); 
}

partial class Test
{
    partial void M(int y){} 
}
```

Reply:
It's limited to a recent warning wave. You can observe the warning in e.g. `PartialMethod_Paramref_02`.